plugin.video.yelo/resources/lib/yelo/yelo.py

```diff
@@ -0,0 +1,311 @@
+            self.save_json_to_file("IdTokens.json", {
+                'deviceId': j["deviceRegistration"]["id"],
+                'webId': j["deviceRegistration"]["deviceProperties"]["dict"][8]["value"]
+            })
+
+    def make_request(self, method, url, headers=None, data=None, json=None, allow_redirects=False, cookies=None,
+                     verify=True):
+        if method.upper() == "POST":
+            r = self.session.post(url=url,
+                                  data=data,
+                                  headers=headers,
+                                  json=json,
+                                  allow_redirects=allow_redirects,
+                                  cookies=cookies,
+                                  verify=verify)
+        elif method.upper() == "GET":
+            r = self.session.get(url=url,
+                                 data=data,
+                                 headers=headers,
+                                 json=json,
+                                 allow_redirects=allow_redirects,
+                                 cookies=cookies,
+                                 verify=verify)
+        else:
+            r = None
+
+        return r
+
+    def login(self):
+        self._prepare_request()
+        self._register_device()
+        self._authorize()
+
+
+        self._login_do()
+
+        callback_url = self._authorize()
+        self._callback(callback_url)
+
+        self._verify_token()
+        self._request_OAuthTokens()
+        self._request_entitlement()
+
+    def _login_do(self):
+        creds = helperclasses.Credentials(self.kodi_wrapper)
+        if not creds.are_filled_in():
+            self.kodi_wrapper.dialog_ok("Login", "You need to fill in your credentials in settings.")
+            self.kodi_wrapper.open_settings()
+            creds.reload()
+
+        self.make_request("POST", "https://login.prd.telenet.be/openid/login.do",
+                          form_headers, create_login_payload(creds.username, creds.password),
+                          None, False, None, self.testing)
+
+    def _callback(self, url):
+        callbackKey = regex(r"(?<=code=)\w{0,32}", url)
+        self.save_json_to_file("callbackKey.json", {"callbackKey": callbackKey})
+
+        self.make_request("GET", url, default_headers, None, None, False, None, self.testing)
+
+    def _verify_token(self):
+        Ids = self.fetch_Ids()
+        callbackKey = self.fetch_CallbackKey()
+        self.make_request("POST", "https://api.yeloplay.be/api/v1/device/verify",
+                          verify_header(Ids["deviceId"], callbackKey), None,
+                          json_verify_header(Ids["deviceId"], Ids["webId"]),
+                          False, None, self.testing)
+
+    def open_json_from_file(self, file_name):
+        path = self.kodi_wrapper.get_addon_data_path()
+
+        if not os.path.exists(path):
+            os.mkdir(path, 0775)
```

Comment:
This is the deprecated octal syntax, use the new one.
```suggestion
            os.mkdir(path, 0o775)
```